httpx/content.py

```diff
@@ -0,0 +1,162 @@
+import typing
+from json import dumps as json_dumps
+from urllib.parse import urlencode
+
+from .multipart import multipart_encode
+
+RequestData = typing.Union[dict, str, bytes, typing.AsyncIterator[bytes]]
+
+RequestFiles = typing.Dict[
+    str,
+    typing.Union[
+        # file (or str)
+        typing.Union[typing.IO[typing.AnyStr], typing.AnyStr],
+        # (filename, file (or str))
+        typing.Tuple[
+            typing.Optional[str], typing.Union[typing.IO[typing.AnyStr], typing.AnyStr],
+        ],
+        # (filename, file (or str), content_type)
+        typing.Tuple[
+            typing.Optional[str],
+            typing.Union[typing.IO[typing.AnyStr], typing.AnyStr],
+            typing.Optional[str],
+        ],
+    ],
+]
+
+
+class RequestContent:
+    """
+    Base class for request content.
+    Defaults to a "no request body" implementation.
+    """
+
+    def get_headers(self) -> typing.Dict[str, str]:
+        return {}
+
+    def can_rewind(self) -> bool:
+        return True
```

Comment:
What is this property supposed to enable in practice? Does that mean we can `async for chunk in content` again once we've done so at least once? I can't think of better names (`.is_replayable()`? `.can_replay()`? `.can_read_multiple_times`? `.is_isomorphic()`? 🤣), but maybe a docstring on this base implementation could help.

Reply:
> Does that mean we can async for chunk in content again

Exactly, yes. It is used when handling redrect responses, that need to generate another request with a body. It happens to be true in all cases except if you pass `data=<async iterator>` in which case we're not able to replay/rewind the request.

And yes, a decent docstring there would make sense.

Reply:
I've updated this to `can_replay` and added a proper docstring.